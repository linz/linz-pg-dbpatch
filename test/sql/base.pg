\set ECHO none
--------------------------------------------------------------------------------
--
-- postgresql-dbpatch - PostgreSQL database patch change management extension
--
-- Copyright 2018 Crown copyright (c)
-- Land Information New Zealand and the New Zealand Government.
-- All rights reserved
--
-- This software is released under the terms of the new BSD license.  See the
-- LICENSE file for more information.
--
--------------------------------------------------------------------------------
-- Provide unit testing for dbpatch using pgTAP
--------------------------------------------------------------------------------

\i test/sql/preparedb

BEGIN;

SELECT plan(5);

SELECT is(apply_patch('test patch 1', ARRAY['SELECT 1', 'SELECT 2']),
  't', 'array of string patches');
SELECT is(apply_patch('test patch 2', 'SELECT 1'),
  't', 'string patch');
SELECT is(apply_patch('test patch 2', 'SELECT 1'),
  'f', 'string patch already applied');
SELECT throws_like($$SELECT apply_patch('test bad patch SQL', 'SELET 1')$$,
  '%bad patch SQL%', 'bad patch syntax throws');

SELECT has_function( 'dbpatch_version' );

ROLLBACK;

