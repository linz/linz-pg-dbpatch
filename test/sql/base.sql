\set VERBOSITY terse
BEGIN;

CREATE EXTENSION dbpatch;

SELECT apply_patch('test patch 1', ARRAY['SELECT 1', 'SELECT 2']);
SELECT apply_patch('test patch 2', 'SELECT 1');
SELECT apply_patch('test patch 2', 'SELECT 1');
SELECT apply_patch('test bad patch SQL', 'SELET 1');

ROLLBACK;

