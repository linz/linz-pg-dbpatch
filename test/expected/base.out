\set VERBOSITY terse
BEGIN;
CREATE EXTENSION dbpatch;
SELECT apply_patch('test patch 1', ARRAY['SELECT 1', 'SELECT 2']);
INFO:  Applying patch test patch 1
 apply_patch 
-------------
 t
(1 row)

SELECT apply_patch('test patch 2', 'SELECT 1');
INFO:  Applying patch test patch 2
 apply_patch 
-------------
 t
(1 row)

SELECT apply_patch('test patch 2', 'SELECT 1');
INFO:  Patch test patch 2 is already applied
 apply_patch 
-------------
 f
(1 row)

SELECT apply_patch('test bad patch SQL', 'SELET 1');
INFO:  Applying patch test bad patch SQL
ERROR:  Could not applied test bad patch SQL patch using SELET 1. ERROR: syntax error at or near "SELET"
ROLLBACK;
